pkg/controller/storagecluster/ensure_Externalstorageclusterresources.go

```diff
@@ -0,0 +1,160 @@
+        Name string            `json:"name"`
+}
+
+// ensureExternalStorageClusterResources ensures that requested resources for the external cluster
+// being created
+func (r *ReconcileStorageCluster) ensureExternalStorageClusterResources(instance *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+        // check for the status boolean value accepted or not
+        if instance.Status.ExternalSecretFound {
+                return nil
+        }
+        found := &corev1.Secret{
+                ObjectMeta: metav1.ObjectMeta{
+                        Name:      externalClusterDetailsSecret,
+                        Namespace: instance.Namespace,
+                },
+        }
+        err := r.client.Get(context.TODO(), types.NamespacedName{Name: found.Name, Namespace: found.Namespace}, found)
+        if err != nil {
+                return err
+        }
+        ownerRef := metav1.OwnerReference{
+                UID:        instance.UID,
+                APIVersion: instance.APIVersion,
+                Kind:       instance.Kind,
+                Name:       instance.Name,
+        }
+        var data []ExternalResource
+        err = json.Unmarshal(found.Data[externalClusterDetailsKey], &data)
+        if err != nil {
+                reqLogger.Error(err, "could not parse json blob")
+                return err
+        }
+        scs, err := r.newStorageClasses(instance)
+        if err != nil {
+                reqLogger.Error(err, "failed to create StorageClasses")
+                return err
+        }
+        r.switchcase_ExternalStorageClusterResources(ownerRef, scs, data, instance, reqLogger)
+        // creating all the storageClasses once we set the values
+        err = r.createStorageClasses(scs, reqLogger)
+        if err != nil {
+                return err
+        }
+        instance.Status.ExternalSecretFound = true
+        return nil
+}
+
+func (r *ReconcileStorageCluster) switchcase_ExternalStorageClusterResources(ownerRef metav1.OwnerReference, scs []*storagev1.StorageClass, data []ExternalResource, instance *ocsv1.StorageCluster, reqLogger logr.Logger) error { 
+       for _, d := range data {
+                objectMeta := metav1.ObjectMeta{
+                        Name:            d.Name,
+                        Namespace:       instance.Namespace,
+                        OwnerReferences: []metav1.OwnerReference{ownerRef},
+                }
+                objectKey := types.NamespacedName{Name: d.Name, Namespace: instance.Namespace}
+                switch d.Kind {
+                case "ConfigMap":
+                        cm := &corev1.ConfigMap{
+                                ObjectMeta: objectMeta,
+                                Data:       d.Data,
+                        }
+                        found := &corev1.ConfigMap{ObjectMeta: objectMeta}
+                        err := r.client.Get(context.TODO(), objectKey, found)
+                        r.create_configmap(err, cm, reqLogger)     
```

Comment:
Don't pass errors into helper functions! :stuck_out_tongue_closed_eyes: Does the complexity go to high if you leave the `err != nil` stanza here?